### PR TITLE
new: RPC schema Unit Test

### PR DESCRIPTION
**Pull request by sanchaymittal**
closes: #255

### CODE REVIEW

File: modules/engine/src/testing/index.spec.ts [hunk 1]
```diff
@@ -69,9 +81,33 @@ describe("VectorEngine", () => {
...
+        timeout: DEFAULT_CHANNEL_TIMEOUT.toString(),
+      });
+      const res = await engine.request(rpc);
+
+      console.log(res.message);
+      expect(res.message).to.contain("fail");
+    });
+    it("should fail if vector.setup fails", async () => {
+      // vectorSetup.setup.resolves(Result.fail(new Error("fail")));
+    });
```
Comment (LayneHaber) [inline modules/engine/src/testing/index.spec.ts:110]: This test should either be skipped or implemented, but shouldn't appear to be passing if nothing is happening

File: modules/engine/src/testing/index.spec.ts [hunk 2]
```diff
@@ -69,9 +81,33 @@ describe("VectorEngine", () => {
...
+      );
+
+      const rpc = constructRpcRequest<"chan_setup">("chan_setup", {
+        counterpartyIdentifier: counterpartyIdentifier,
+        chainId: chainId,
+        timeout: DEFAULT_CHANNEL_TIMEOUT.toString(),
+      });
+      const res = await engine.request(rpc);
+
+      console.log(res.message);
```
Comment (LayneHaber) [inline modules/engine/src/testing/index.spec.ts:105]: can remove

File: modules/engine/src/testing/index.spec.ts [hunk 3]
```diff
@@ -90,18 +126,715 @@ describe("VectorEngine", () => {
...
+        error: "should be equal to one of the allowed values",
+      },
+      {
+        name: "no request.method",
+        overrides: { id: 1, jsonrpc: "2.0", method: undefined, params: undefined },
+        error: "should have required property 'method'",
+      },
+      {
+        name: "malformed request.method",
+        overrides: { id: 1, jsonrpc: "2.0", method: "invalid_method", params: undefined },
```
Comment (LayneHaber) [inline modules/engine/src/testing/index.spec.ts:163]: hmmm, i dont think this should succeed if the `params` can be `undefined`. i believe it should be an empty body cc @rhlsthrm

File: modules/engine/src/testing/index.spec.ts [hunk 4]
```diff
@@ -90,18 +126,715 @@ describe("VectorEngine", () => {
...
+          Sinon.createStubInstance(VectorChainService),
+          chainAddresses,
+          log,
+        );
+        let rpc: EngineParams.RpcRequest = { ...test.overrides } as any;
+        try {
+          await engine.request(rpc);
+        } catch (res) {
+          expect(res.message).to.contain(test.error);
+        }
```
Comment (LayneHaber) [inline modules/engine/src/testing/index.spec.ts:188]: can replace with `await expect(engine.request(rpc)).rejectedWith(test.error)`

File: modules/engine/src/testing/index.spec.ts [hunk 5]
```diff
@@ -90,18 +126,715 @@ describe("VectorEngine", () => {
...
+            Sinon.createStubInstance(VectorChainService),
+            chainAddresses,
+            log,
+          );
+          let rpc: EngineParams.RpcRequest = { id: 1, jsonrpc: "2.0", ...test.overrides } as any;
+          try {
+            await engine.request(rpc);
+          } catch (res) {
+            expect(res.message).to.be.eq(test.error);
+          }
```
Comment (LayneHaber) [inline modules/engine/src/testing/index.spec.ts:835]: See above replacement suggestion

**Review by LayneHaber — COMMENTED** [review]
Leaving this as unapproved until I hear back about the empty params in the rpc schema from @rhlsthrm

File: modules/engine/src/testing/index.spec.ts (hunk 3 above)
Comment (rhlsthrm) [reply modules/engine/src/testing/index.spec.ts:163]: Some of the RPC schema is actually undefined. Maybe we should change it to be empty body only, I'm okay with that.

**Review by LayneHaber — CHANGES_REQUESTED** [review]
Should make the `rpc.params` not allowed to be undefined (i.e. empty body instead)

File: modules/engine/src/testing/index.spec.ts [hunk 6]
```diff
@@ -69,9 +81,33 @@ describe("VectorEngine", () => {
   // complex logic within the index file of the module)
 
   // NOTE: these are nice-to-haves
+
   describe.skip("setup", () => {
```
[the remark this answers is not in the document]
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts:85]: Still need to complete setup and withdrawal test. Will create another pr for it.

File: modules/engine/src/testing/index.spec.ts (hunk 1 above)
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts:110]: Skipped, for now, removing commented code too.

File: modules/engine/src/testing/index.spec.ts (hunk 2 above)
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts:105]: removed console.log

File: modules/engine/src/testing/index.spec.ts (hunk 4 above)
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts:188]: Ohh noice!

File: modules/engine/src/testing/index.spec.ts (hunk 5 above)
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts:835]: Done

File: modules/engine/src/testing/index.spec.ts (hunk 3 above)
Comment (sanchaymittal) [reply modules/engine/src/testing/index.spec.ts]: Done

**Review by LayneHaber — APPROVED** [review]

**Pull request merged**